### PR TITLE
YOMA-520 - feat(UserSkills): store user skills credentials in state

### PR DESCRIPTION
**Pull request by sciphergfx**
## Work Done
- redux setup for userSkills

### CODE REVIEW

File: src/modules/UserSkills/UserSkills.constants.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+export const FALLBACK_SKILLS_LENGTH = 20
+export const SEARCH_TERM_MIN_LENGTH = 3
+
+export const MOCK_SKILLS = [
```
Comment (cameronolivier) [inline src/modules/UserSkills/UserSkills.constants.ts:1]: suggestion: we should delete the mock skills - I dont believe we need them anymore (do a usage check to confirm)

File: src/modules/Skills/Skills.constants.ts [hunk 2]
```diff
@@ -1,25 +1,2 @@
 export const FALLBACK_SKILLS_LENGTH = 20
```
Comment (cameronolivier) [inline src/modules/Skills/Skills.constants.ts:1]: issue: it seems we're repeating these two values - here and in the UserSkills constants

File: src/modules/UserSkills/UserSkillsForm/UserSkillsForm.constants.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
-import { SkillsField } from './SkillsForm.types'
+import { UserSkillsField } from './UserSkillsForm.types'
 
-export const SKILLS_INITIAL_VALUES: SkillsField = {
+export const USER_SKILLS_INITIAL_VALUES: UserSkillsField = {
   skillNames: [],
 }
 
-export const MOCK_SKILLS_LIST = [
+export const MOCK_USER_SKILLS_LIST = [
```
Comment (cameronolivier) [inline src/modules/UserSkills/UserSkillsForm/UserSkillsForm.constants.ts:7]: suggestion: we should delete this if we aren't using it (And we shouldn't be using it anymore).

File: src/modules/UserSkills/UserSkillsForm/UserSkillsForm.tsx [hunk 4]
```diff
@@ -4,18 +4,18 @@ import React from 'react'
...
   const { t } = useTranslation()
 
   return (
-    <Formik initialValues={SKILLS_INITIAL_VALUES} onSubmit={() => {}}>
+    <Formik initialValues={USER_SKILLS_INITIAL_VALUES} onSubmit={() => {}}>
       {(formikHandlers: FormikProps<FormikValues>) => (
         <View style={styles.form}>
           <DropDownTags
-            items={MOCK_SKILLS_LIST}
+            items={MOCK_USER_SKILLS_LIST}
```
Comment (cameronolivier) [inline src/modules/UserSkills/UserSkillsForm/UserSkillsForm.tsx:18]: we should be using real values going forward.

**Review by cameronolivier — CHANGES_REQUESTED** [review]

File: src/modules/UserSkills/UserSkillsForm/UserSkillsForm.tsx (hunk 4 above)
Comment (sciphergfx) [reply src/modules/UserSkills/UserSkillsForm/UserSkillsForm.tsx:18]: considering the ticket is  storing user skills credentials in state don't think this is important

File: src/modules/UserSkills/UserSkillsForm/UserSkillsForm.constants.ts (hunk 3 above)
Comment (sciphergfx) [reply src/modules/UserSkills/UserSkillsForm/UserSkillsForm.constants.ts:7]: considering the ticket is storing user skills credentials in state don't think this is important.

File: src/modules/UserSkills/UserSkillsForm/UserSkillsForm.tsx (hunk 4 above)
Comment (sciphergfx) [reply src/modules/UserSkills/UserSkillsForm/UserSkillsForm.tsx:18]: there's a ticket that handles this

**Review by cameronolivier — APPROVED** [review]

**Pull request merged**